File: intelligencecity_manager_web/src/main/webapp/assets/js/service/deviceService.js
//服务层
app.service('deviceService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../device/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../device/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../device/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../device/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../device/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../device/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../device/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: intelligencecity_manager_interface/src/main/java/com/shanghaiuniversity/manager/service/DeviceService.java
package com.shanghaiuniversity.manager.service;

import com.ShanghaiUniversity.entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface DeviceService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<Device> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加
     */
    public void add(Device device);

    /**
     * 修改
     */
    public void update(Device device);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public Device findOne(int id);

    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(int[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Device device, int pageNum, int pageSize);

}
